homeassistant/components/lutron/light.py

```diff
@@ -33,36 +35,50 @@ async def async_setup_entry(
     """
     ent_reg = er.async_get(hass)
     entry_data: LutronData = hass.data[DOMAIN][config_entry.entry_id]
+    lights = []
 
     for area_name, device in entry_data.lights:
         if device.type == "CEILING_FAN_TYPE":
             # If this is a fan check to see if this entity already exists. If not, do not create a new one.
-            entity_id = f"light.{slugify(f'{area_name} {device.name}')}"
-            if ent_reg.async_is_registered(entity_id):
+            entity_id = ent_reg.async_get_entity_id(
+                Platform.LIGHT,
+                DOMAIN,
+                f"{entry_data.client.guid}_{device.uuid}",
+            )
+            if entity_id:
                 entity_entry = ent_reg.async_get(entity_id)
                 assert entity_entry
                 if entity_entry.disabled:
                     # If the entity exists and is disabled then we want to remove the entity so that the user is using the new fan entity instead.
                     ent_reg.async_remove(entity_id)
                 else:
-                    async_add_entities(
-                        [LutronLight(area_name, device, entry_data.client)],
-                        True,
-                    )
-                    async_create_issue(
-                        hass,
-                        DOMAIN,
-                        "deprecated_light_fan",
-                        breaks_in_ha_version="2024.7.0",
-                        is_fixable=False,
-                        severity=IssueSeverity.WARNING,
-                        translation_key="deprecated_light_fan",
-                    )
+                    lights.append([area_name, device, entry_data.client])
+                    entity_automations = automations_with_entity(hass, entity_id)
+                    entity_scripts = scripts_with_entity(hass, entity_id)
+                    for item in entity_automations + entity_scripts:
+                        async_create_issue(
+                            hass,
+                            DOMAIN,
+                            f"deprecated_light_fan_{entity_id}_{item}",
+                            breaks_in_ha_version="2024.8.0",
+                            is_fixable=False,
```

Comment:
Should be fixable and persistent so the user can fix the issue when they have updated the automations and scripts.